### PR TITLE
AL-1641 Support SSM Docs in Deployment

### PR DESCRIPTION
**Pull request by jconstance-amplify**
Added support for toggling testing mode and executing integration tests
via SSM documents instead of through SSH. This new behavior is opt-in
and requires two new configuration options be set. Details in the
README.

Open to thoughts on the interface for the SSM documents specified in the README.

### CODE REVIEW

File: disco_aws_automation/disco_deploy.py [hunk 1]
```diff
@@ -519,6 +521,19 @@ def _set_testing_mode(self, hostclass, instances, mode_on):
...
+                document_name=ssm_doc_testing_mode,
+                parameters={
+                    "mode": "on" if mode_on else "off"
+                },
+                comment="Asiaq Deploy is toggling testing mode"
+            )
+
         exit_code = 0
         for inst in instances:
             _code, _stdout = self._disco_aws.remotecmd(
```
Comment (aoskotsky-amplify) [inline disco_aws_automation/disco_deploy.py]: Shouldn't this be removed?

Comment (jconstance-amplify) [inline disco_aws_automation/disco_deploy.py]: That would not be backwards compatible.

Comment (aoskotsky-amplify) [inline disco_aws_automation/disco_deploy.py]: Maybe an if statement then to only use SSH if there is no SSM doc?

Comment (jconstance-amplify) [inline disco_aws_automation/disco_deploy.py]: That's what we essentially have here already, no?

**Pull request merged**